src/bridge/transactions/kick_off_timeout.rs

```diff
@@ -0,0 +1,194 @@
+    fn musig2_nonces_mut(&mut self) -> &mut HashMap<usize, HashMap<PublicKey, PubNonce>> {
+        &mut self.musig2_nonces
+    }
+    fn musig2_nonce_signatures(&self) -> &HashMap<usize, HashMap<PublicKey, Signature>> {
+        &self.musig2_nonce_signatures
+    }
+    fn musig2_nonce_signatures_mut(
+        &mut self,
+    ) -> &mut HashMap<usize, HashMap<PublicKey, Signature>> {
+        &mut self.musig2_nonce_signatures
+    }
+    fn musig2_signatures(&self) -> &HashMap<usize, HashMap<PublicKey, PartialSignature>> {
+        &self.musig2_signatures
+    }
+    fn musig2_signatures_mut(
+        &mut self,
+    ) -> &mut HashMap<usize, HashMap<PublicKey, PartialSignature>> {
+        &mut self.musig2_signatures
+    }
+}
+
+impl KickOffTimeoutTransaction {
+    pub fn new(context: &OperatorContext, input_0: Input) -> Self {
+        Self::new_for_validation(
+            context.network,
+            &context.operator_taproot_public_key,
+            &context.n_of_n_taproot_public_key,
+            input_0,
+        )
+    }
+
+    pub fn new_for_validation(
+        network: Network,
+        operator_taproot_public_key: &XOnlyPublicKey,
+        n_of_n_taproot_public_key: &XOnlyPublicKey,
+        input_0: Input,
+    ) -> Self {
+        let connector_1 = Connector1::new(
+            network,
+            &operator_taproot_public_key,
+            &n_of_n_taproot_public_key,
+        );
+
+        let input_0_leaf = 1;
+        let _input_0 = connector_1.generate_taproot_leaf_tx_in(input_0_leaf, &input_0);
+
+        let total_output_amount = input_0.amount - Amount::from_sat(FEE_AMOUNT);
+
+        let _output_0 = TxOut {
+            value: total_output_amount * 95 / 100,
+            script_pubkey: generate_burn_script_address(network).script_pubkey(),
+        };
+
+        let reward_output_amount = total_output_amount - (total_output_amount * 95 / 100);
+        let _output_1 = TxOut {
+            value: reward_output_amount,
+            script_pubkey: ScriptBuf::default(),
+        };
+
+        KickOffTimeoutTransaction {
+            tx: Transaction {
+                version: bitcoin::transaction::Version(2),
+                lock_time: absolute::LockTime::ZERO,
+                input: vec![_input_0],
+                output: vec![_output_0, _output_1],
+            },
+            prev_outs: vec![TxOut {
+                value: input_0.amount,
+                script_pubkey: connector_1.generate_taproot_address().script_pubkey(),
+            }],
+            prev_scripts: vec![connector_1.generate_taproot_leaf_script(input_0_leaf)],
+            connector_1,
+            reward_output_amount,
+            musig2_nonces: HashMap::new(),
+            musig2_nonce_signatures: HashMap::new(),
+            musig2_signatures: HashMap::new(),
+        }
+    }
+
+    pub fn num_blocks_timelock_0(&self) -> u32 { self.connector_1.num_blocks_timelock_1 }
```

Comment:
Is this fn used anywahere?

Reply:
It should be in the calculation of its status

Reply:
Line 843 graphs/oeg_out